lib/fc.js

```diff
@@ -208,13 +208,23 @@ async function makeFunction(baseDir, {
     }
   }
 
-  if (!fn) {
-    // create
-    params['functionName'] = functionName;
-    fn = await fc.createFunction(serviceName, params);
-  } else {
-    // update
-    fn = await fc.updateFunction(serviceName, functionName, params);
+  try {
+
+    if (!fn) {
+      // create
+      params['functionName'] = functionName;
+      fn = await fc.createFunction(serviceName, params);
+    } else {
+      // update
+      fn = await fc.updateFunction(serviceName, functionName, params);
+    }
+  } catch (ex) {
+
+    if (ex.message.indexOf('timeout') !== -1) {
+
+      throw new Error('\ndeploy failed: may be caused by network timeout. you can try the following ways:\n1.increase the timeout by "fun config".\n2.add .funignore file ignores useless files or use webpack to tailor the package size and redeploy it.');
```

Comment:
Deploy function failed. 
Error message is ${ex.message}.

This may be caused by network latency. You can set the client timeout to a larger value through `fun config` and then try again.

Reply:
不用提及 .funignore 和 webpack 吗

Reply:
funignore 可以加上来，webpack 只是针对 node 的吧